lib/features/modeling/cmd/LinkCallChoreoParticipantHandler.js

```diff
@@ -0,0 +1,55 @@
+export default function LinkCallChoreoParticipantHandler(moddle) {
+  this._moddle = moddle;
+}
+
+/**
+ * Links the outer Participant of a Call Choreo to the inner Particiant of called Chore by creating a participant
+ * association. If the inner Participant is undefined it works as an unlink.
+ * @param element
+ * @param outerParticipant
+ * @param innerParticipant
+ * @returns {*[]}
+ */
+LinkCallChoreoParticipantHandler.prototype.link = function(element, outerParticipant, innerParticipant) {
+  const oldParticipantAssociation = element.participantAssociations.find(
+    pa => pa.outerParticipantRef === outerParticipant);
+
+  if (oldParticipantAssociation) {
+    // We remove the old participant association
+    element.participantAssociations = element.participantAssociations.filter(
+      pa => pa !== oldParticipantAssociation);
+
+    if (oldParticipantAssociation.innerParticipantRef === innerParticipant) {
+      // In case the current innerParticipant is set again, we interpret it as a toggling and remove it.
+      return [element];
+    }
+  }
+  if (innerParticipant) {
+    const participantAssociation = this._moddle.create('bpmn:ParticipantAssociation', {
+      outerParticipantRef: outerParticipant,
+      innerParticipantRef: innerParticipant
+    });
+
+    element.participantAssociations.push(participantAssociation);
+  }
+
+  return [element];
+};
+
+LinkCallChoreoParticipantHandler.prototype.preExecute = function(context) {
+  context.element.participantAssociations = context.element.participantAssociations || [];
+
+  const participantAssociation = context.element.participantAssociations.find(
```

Comment:
Candidate for its own function since the same code is used above in `link` and the popup provider as well.